lua/autorun/server/sv_spawn_protection.lua

```diff
@@ -188,81 +176,80 @@ local function setSpawnProtectionForPvpSpawn( ply )
     setLastSpawnTime( ply )
     setSpawnProtection( ply )
     setPlayerTransparent( ply )
-    setPlayerNoCollide( ply )
     createDecayTimer( ply )
 end
 
+-- Instantly removes spawn protection and removes timers and alpha level.
+local function instantRemoveSpawnProtection( ply, message )
+    if not playerHasSpawnProtection( ply ) then return end 
+    removeSpawnProtection( ply, message )
+    setPlayerVisible( ply )
+    removeDecayTimer( ply )
+    removeDelayedRemoveTimer( ply )
+end
+
 -- Called on weapon change to check if the weapon is allowed,
 -- and remove spawn protection if it's not
-local function spawnProtectionWeaponChangeCheck( ply, oldWeapon, newWeapon )
+local function spawnProtectionWeaponChangeCheck( ply, _, newWeapon )
     if not playerIsInPvp( ply ) then return end
     if not playerHasSpawnProtection( ply ) then return end
     if weaponIsAllowed( newWeapon ) then return end
 
     local lastSpawnTime = ply:GetNWInt( "lastSpawnTime", CurTime() - spawnProtectionWeaponGracePeriod )
     if lastSpawnTime >= CurTime() - spawnProtectionWeaponGracePeriod then return end
 
-    local printMessage = "You've equipped a weapon and lost spawn protection"
-    removeSpawnProtection( ply, printMessage )
-    setPlayerVisible( ply )
-    setPlayerCollide( ply )
-    removeDecayTimer( ply )
-    removeDelayedRemoveTimer( ply )
+    instantRemoveSpawnProtection( ply, "You've equipped a weapon and lost spawn protection." )
+end
+
+-- Remove spawn protection on using objects
+local function spawnProtectionUseCheck( ply )
+    if not playerIsInPvp( ply ) then return end
+    if not playerHasSpawnProtection( ply ) then return end
+    instantRemoveSpawnProtection( ply, "You've used an entity and lost spawn protection." )
 end
 
 -- Called on player keyDown events to check if a movement key was pressed
 -- and remove spawn protection if so
 local function spawnProtectionMoveCheck( ply, keyCode )
     if playerIsDisablingSpawnProtection( ply ) then return end
     if not playerHasSpawnProtection( ply ) then return end
-    if keyVoidsSpawnProtection[ keyCode ] then delayRemoveSpawnProtection( ply ) end
+    if keyVoidsSpawnProtection[ keyCode ] then
+        delayRemoveSpawnProtection( ply )
+    end
```

Comment:
Follows the RERO pattern in this repo
```suggestion
    if not keyVoidsSpawnProtection[ keyCode ] then return end

    delayRemoveSpawnProtection( ply )
```

Reply:
Yeah just style changed it, should've done that too 👍